src/include/OpenImageIO/fmath.h

```diff
@@ -214,9 +214,9 @@ OIIO_FORCEINLINE uint64_t rotl64 (uint64_t x, int k) {
 
 
 /// clamp a to bounds [low,high].
-template <class T>
+template <class T, class S>
 inline T
-clamp (T a, T low, T high)
+clamp (T a, S low, S high)
```

Comment:
I'm not sure I follow what situation would lead to more type errors with all arguments being of the same type. It seems its mixing types that would lead to trouble, for example:

`size_t clamp(size_t, int ,int)` would complain about comparing signed and unsigned?